oscore-cf/src/main/java/org/eclipse/californium/oscore/HashMapCtxDB.java

```diff
@@ -0,0 +1,236 @@
+/*******************************************************************************
+ * Copyright (c) 2018 RISE SICS and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Joakim Brorsson
+ *    Ludwig Seitz (RISE SICS)
+ *    Tobias Andersson (RISE SICS)
+ *    
+ ******************************************************************************/
+package org.eclipse.californium.oscore;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.logging.Logger;
+
+import org.eclipse.californium.core.coap.CoAP.Code;
+import org.eclipse.californium.core.coap.OptionSet;
+import org.eclipse.californium.core.coap.Token;
+import org.eclipse.californium.core.network.RandomTokenGenerator;
+import org.eclipse.californium.core.network.TokenGenerator;
+import org.eclipse.californium.core.network.config.NetworkConfig;
+
+public class HashMapCtxDB implements OSCoreCtxDB {
+
+	/**
+	 * The logger
+	 */
+	private static final Logger LOGGER = Logger.getLogger(HashMapCtxDB.class.getName());
+	private HashMap<ByteId, OSCoreCtx> ridMap;
```

Comment:
Please read the comment on `ByteId.hashCode()`!

Reply:
Done.